pkg/gds/gds.go

```diff
@@ -71,23 +71,28 @@ func (s *GDS) Serve() (err error) {
 	}
 
 	// Run the server
-	go func() {
-		defer sock.Close()
-		log.Info().
-			Str("listen", s.conf.BindAddr).
-			Str("version", pkg.Version()).
-			Msg("trisa directory server started")
-
-		if err := s.srv.Serve(sock); err != nil {
-			s.svc.echan <- err
-		}
-	}()
+	go s.Run(sock)
 
 	// Now that the go routine is started return nil, meaning the service has started
 	// successfully with no problems.
 	return nil
 }
 
+// Run the gRPC server. This method is extracted from the Serve function so that it can
+// be run in its own go routine and to allow tests to Run a bufconn server without
+// starting a live server with all of the various go routines and channels running.
+func (s *GDS) Run(sock net.Listener) {
+	defer sock.Close()
+	log.Info().
+		Str("listen", s.conf.BindAddr).
+		Str("version", pkg.Version()).
+		Msg("trisa directory server started")
+
+	if err := s.srv.Serve(sock); err != nil {
+		s.svc.echan <- err
+	}
+}
+
```

Comment:
This makes me nervous but hopefully it should be obvious if it breaks anything?

Reply:
It looks fine to me! Would you mind moving the log statement back to the `Serve` method? I think it makes a bit more sense there and I believe I did the same thing in trtl. I've added a suggestion to make it easy.